application/client/src/app/ui/views/toolbar/chart/chart/state.ts

```diff
@@ -39,8 +45,54 @@ export class State extends AdvancedState {
     }
 
     public onClick(event: MouseEvent) {
+        const position: number = this._calculatePosition(event);
+        position >= 0 && this._session.cursor.select(position, Owner.Chart);
+    }
+
+    public onMouseMove(event: MouseEvent) {
+        const rect: DOMRect = this._element.getBoundingClientRect();
```

Comment:
`getBoundingClientRect` is a very-very expensive operation. That's not a good idea to do it with `mousemove` event.